src/app/clusters/scenes/SceneTableImpl.cpp

```diff
@@ -0,0 +1,680 @@
+        ReturnErrorOnFailure(reader.Get(scene_count));
+        for (uint8_t i = 0; i < kMaxScenePerFabric; i++)
+        {
+            ReturnErrorOnFailure(reader.Next(TLV::ContextTag(kTagSceneEndpointID)));
+            ReturnErrorOnFailure(reader.Get(scene_map[i].mEndpointId));
+            ReturnErrorOnFailure(reader.Next(TLV::ContextTag(kTagSceneGroupID)));
+            ReturnErrorOnFailure(reader.Get(scene_map[i].mGroupId));
+            ReturnErrorOnFailure(reader.Next(TLV::ContextTag(kTagSceneID)));
+            ReturnErrorOnFailure(reader.Get(scene_map[i].mSceneId));
+        }
+        ReturnErrorOnFailure(reader.Next(TLV::ContextTag(kTagNext)));
+        ReturnErrorOnFailure(reader.Get(next));
+
+        return reader.ExitContainer(container);
+    }
+
+    // Register the fabric in the list of fabrics having scenes
+    CHIP_ERROR Register(PersistentStorageDelegate * storage)
+    {
+        FabricHavingSceneList fabric_list;
+        CHIP_ERROR err = fabric_list.Load(storage);
+        if (CHIP_ERROR_NOT_FOUND == err)
+        {
+            // New fabric list
+            fabric_list.first_entry = fabric_index;
+            fabric_list.entry_count = 1;
+            return fabric_list.Save(storage);
+        }
+        ReturnErrorOnFailure(err);
+
+        // Existing fabric list, search for existing entry
+        FabricSceneData fabric(fabric_list.first_entry);
+        for (size_t i = 0; i < fabric_list.entry_count; i++)
+        {
+            err = fabric.Load(storage);
+            if (CHIP_NO_ERROR != err)
+            {
+                break;
+            }
+            if (fabric.fabric_index == this->fabric_index)
+            {
+                // Fabric already registered
+                return CHIP_NO_ERROR;
+            }
+            fabric.fabric_index = fabric.next;
+        }
+        // Add this fabric to the fabric list
+        this->next              = fabric_list.first_entry;
+        fabric_list.first_entry = this->fabric_index;
+        fabric_list.entry_count++;
+        return fabric_list.Save(storage);
+    }
+
+    // Remove the fabric from the fabrics' linked list
+    CHIP_ERROR Unregister(PersistentStorageDelegate * storage) const
+    {
+        FabricHavingSceneList fabric_list;
+        CHIP_ERROR err = fabric_list.Load(storage);
+        VerifyOrReturnError(CHIP_NO_ERROR == err || CHIP_ERROR_NOT_FOUND == err, err);
+
+        // Existing fabric list, search for existing entry
+        FabricSceneData fabric(fabric_list.first_entry);
+        FabricSceneData prev;
+
+        for (size_t i = 0; i < fabric_list.entry_count; i++)
+        {
+            err = fabric.Load(storage);
+            if (CHIP_NO_ERROR != err)
+            {
+                break;
+            }
+            if (fabric.fabric_index == this->fabric_index)
+            {
+                // Fabric found
+                if (i == 0)
+                {
+                    // Remove first fabric
```

Comment:
Hold on a sec.  We are using the same list for both scenes and groups, no?  Why is it ok to remove things here?  They might still have groups defined, no?

This needs to be very clearly explained if it's actually correct, but it does not look correct to me.

Reply:
FabricList removed from scenes, the Fabric Table is already capable of retrieving fabric indexes and from there the scene table can access a FabricSceneData.